dev/PushNotifications/PushNotificationsLongRunningTask/NotificationListenerManager.cpp

```diff
@@ -15,30 +18,31 @@ void NotificationListenerManager::Initialize(std::shared_ptr<ForegroundSinkManag
 
 void NotificationListenerManager::AddListener(std::wstring appId, std::wstring processName)
 {
-    auto lock = m_lock.lock_exclusive();
-
     THROW_HR_IF(E_INVALIDARG, appId.empty());
     THROW_HR_IF(E_INVALIDARG, processName.empty());
 
-    if (m_notificationListeners.find(appId) == std::end(m_notificationListeners))
-    {
-        ComPtr<INotificationListener> listener;
-        THROW_IF_FAILED(MakeAndInitialize<NotificationListener>(&listener, m_foregroundSinkManager, appId, processName));
-        THROW_IF_FAILED(PushNotifications_RegisterFullTrustApplication(appId.c_str(), GUID_NULL));
-        THROW_IF_FAILED(PushNotifications_RegisterNotificationSinkForFullTrustApplication(appId.c_str(), listener.Get()));
-
-        AgileRef agileListener;
-        THROW_IF_FAILED(AsAgile(listener.Get(), &agileListener));
-        m_notificationListeners.insert({ appId, agileListener });
-    }
+    auto lock = m_lock.lock_exclusive();
+
+    // Make sure we keep the long running sink up-to-date with wpncore.
+    ComPtr<INotificationListener> newListener;
+    THROW_IF_FAILED(MakeAndInitialize<NotificationListener>(&newListener, m_foregroundSinkManager, appId, processName));
+    THROW_IF_FAILED(PushNotifications_RegisterNotificationSinkForFullTrustApplication(appId.c_str(), newListener.Get()));
```

Comment:
Not sure, Just wondering why we don't need to  call "THROW_IF_FAILED(PushNotifications_RegisterFullTrustApplication(appId.c_str(), GUID_NULL))" as the original code does?

Reply:
Right. If the registration does not exist, this call to register a sink will likely fail. What was the reasoning for deleting that here?

Reply:
It's a mistake. Thanks for catching this!